MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/javaservices/ElevationQueryActivity.java

```diff
@@ -0,0 +1,285 @@
+          // elevation number
+          if (featureList.size() > 0) {
+            GeoJsonSource resultSource = style.getSourceAs(RESULT_GEOJSON_SOURCE_ID);
+            if (resultSource != null) {
+              resultSource.setGeoJson(featureList.get(featureList.size() - 1));
+            }
+          }
+        } else {
+          String noFeaturesString = getString(R.string.elevation_tilequery_no_features);
+          Timber.d(noFeaturesString);
+          Toast.makeText(ElevationQueryActivity.this, noFeaturesString, Toast.LENGTH_SHORT).show();
+          elevationQueryNumbersOnlyResponseTextView.setText(noFeaturesString);
+          elevationQueryJsonResponseTextView.setText(noFeaturesString);
+        }
+      }
+
+      @Override
+      public void onFailure(Call<FeatureCollection> call, Throwable throwable) {
+        Timber.d("Request failed: %s", throwable.getMessage());
+        Toast.makeText(ElevationQueryActivity.this,
+          R.string.elevation_tilequery_api_response_error, Toast.LENGTH_SHORT).show();
+      }
+    });
+  }
+
+  /**
+   * Use the Maps SDK's LocationComponent to display the device location on the map
+   */
+  @SuppressWarnings( {"MissingPermission"})
+  private void displayDeviceLocation(@NonNull Style loadedMapStyle) {
+    // Check if permissions are enabled and if not request
+    if (PermissionsManager.areLocationPermissionsGranted(this)) {
+
+      // Get an instance of the component
+      LocationComponent locationComponent = mapboxMap.getLocationComponent();
+
+      // Activate with options
+      locationComponent.activateLocationComponent(
+        LocationComponentActivationOptions.builder(this, loadedMapStyle).build());
+
+      // Enable to make component visible
+      locationComponent.setLocationComponentEnabled(true);
+
+      // Set the component's camera mode
+      locationComponent.setCameraMode(CameraMode.TRACKING);
+      locationComponent.setRenderMode(RenderMode.COMPASS);
+
+      // Zoom the camera into the device's current location
+      mapboxMap.animateCamera(CameraUpdateFactory
+        .newCameraPosition(new CameraPosition.Builder()
+          .zoom(10)
+          .build()), 2000);
+    } else {
+      permissionsManager = new PermissionsManager(this);
+      permissionsManager.requestLocationPermissions(this);
+    }
+  }
+
+  // The following three methods are related to showing the device's location via the LocationComponent
+  @Override
+  public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
+    permissionsManager.onRequestPermissionsResult(requestCode, permissions, grantResults);
+  }
+
+  @Override
+  public void onExplanationNeeded(List<String> permissionsToExplain) {
+    Toast.makeText(this, R.string.user_location_permission_explanation, Toast.LENGTH_LONG).show();
+  }
+
+  @Override
+  public void onPermissionResult(boolean granted) {
+    Style style = mapboxMap.getStyle();
```

Comment:
Same as above, shall we use the async callback here for the `getStyle()`?

Reply:
Sure. Fixed